spec/unit/provider/exec/powershell_spec.rb

```diff
@@ -115,6 +116,19 @@
       }
     MANIFEST
     }
+    let(:tmpdir) { Dir.mktmpdir('statetmp').encode!(Encoding::UTF_8) }
+
+    before :each do
+      # a statedir setting must now exist per the new transactionstore code
+      # introduced in Puppet 4.6 for corrective changes, as a new YAML file
+      # called transactionstore.yaml will be written under this path
+      # which defaults to c:\dev\null when not set on Windows
+      Puppet[:statedir] = tmpdir
+    end
+
+    after :each do
+      FileUtils.rm_rf(tmpdir)
+    end
```

Comment:
This seems better to change as part of the spec helper (and then later as puppetlabs_spec_helper).